crates/bitcoin-da/src/service.rs

```diff
@@ -583,9 +583,8 @@ impl DaService for BitcoinService {
             "Extracting relevant txs from block {:?}",
             block.header.block_hash()
         );
-
-        let txs = block.txdata.iter().map(|tx| tx.inner().clone()).collect();
-        get_relevant_blobs_from_txs(txs, &self.reveal_batch_prover_prefix)
+        let txs: Vec<&Transaction> = block.txdata.iter().map(|tx| tx.inner()).collect(); // get a list of references to the transactions
```

Comment:
The trick is to get rid of .collect() if possible. And it is possible

Reply:
😁
don't tell him how tho :p